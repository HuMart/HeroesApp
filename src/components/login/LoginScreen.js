import React, { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { types } from '../../types/types';


export const LoginScreen = () => {

  

  const { dispatch } = useContext(AuthContext);

  const navigate = useNavigate();

  const handleLogin = () => {

    const action = {

      type: types.login,
      payload: { name: 'User Logged' }
    }

    dispatch(action);

    const lastPath = localStorage.getItem('lastPath') || '/marvel'

    navigate(lastPath, {
      replace: true
    });
  }

  return (
    <div  className='position-absolute top-50 start-50 translate-middle'>
      <img className='container  w-auto h-auto opacity-50 ' src='https://wallpapertag.com/wallpaper/full/1/1/1/549111-marvel-vs-dc-wallpaper-1929x1200-for-android-50.jpg' alt=''/>
      <div className='img-overlay text-dark text-center position-absolute top-50 start-50 translate-middle'>
          <h1>Enter to the World of Heroes</h1>
          <hr />

          <button 
            className='btn btn-transparent btn-lg' 
            onClick={ handleLogin }
          >
            Click Me
          </button>
      </div>      
      <footer className='text-center text-white bg-dark'>
        MIT <span className="material-icons-outlined">
copyright
</span>
      </footer>

    </div>

  );
};
