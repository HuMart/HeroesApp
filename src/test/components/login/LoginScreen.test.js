import React from "react";
import { mount } from "enzyme";
import { Routes, MemoryRouter, Route } from "react-router-dom";

import { AuthContext } from "../../../auth/authContext";
import { types } from "../../../types/types";
import { LoginScreen } from "../../../components/login/LoginScreen";



const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}))


describe('Test <LoginScreen />', () => { 

    const contextValue =  {
        dispatch: jest.fn(),
        user: {
            logged: false,
        }
    }    

    const wrapper = mount(
        <AuthContext.Provider value={ contextValue }>
            <MemoryRouter initialEntries={ ['/login'] }>
                <Routes>
                    <Route path="/login" element={<LoginScreen />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>

    )


    test('should render correctly', () => { 

        expect( wrapper ).toMatchSnapshot();

    })

    test('should execute dispatch and navigate to the next screen', () => { 

        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect( contextValue.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload: { name: 'User Logged' }
        });
        expect(mockNavigate).toHaveBeenCalledWith( "/marvel", { replace: true } );

        localStorage.setItem('lastPath', '/dc');

        handleClick();
        expect(mockNavigate).toHaveBeenCalledWith( "/dc", { replace: true } );


    })

 })