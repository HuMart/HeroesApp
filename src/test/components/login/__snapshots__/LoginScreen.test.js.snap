// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <LoginScreen /> should render correctly 1`] = `
<div
  className="position-absolute top-50 start-50 translate-middle"
>
  <img
    alt=""
    className="container  w-auto h-auto opacity-50 "
    src="https://wallpapertag.com/wallpaper/full/1/1/1/549111-marvel-vs-dc-wallpaper-1929x1200-for-android-50.jpg"
  />
  <div
    className="img-overlay text-dark text-center position-absolute top-50 start-50 translate-middle"
  >
    <h1>
      Enter to the World of Heroes
    </h1>
    <hr />
    <button
      className="btn btn-transparent btn-lg"
      onClick={[Function]}
    >
      Click Me
    </button>
  </div>
  <footer
    className="text-center text-white bg-dark"
  >
    MIT 
    <span
      className="material-icons-outlined"
    >
      copyright
    </span>
  </footer>
</div>
`;
